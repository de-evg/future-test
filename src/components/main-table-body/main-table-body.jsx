import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { NameSpace } from "../../store/reducers/root";
import MainTableRow from "../main-table-row/main-table-row";
import { ActionCreator } from "../../store/action";
import { getFilteredUsers } from "../../store/selectors";
import { STEP } from "../../const";

const MainTableBody = ({
  users,
  activeRow,
  setActiveRow,
  currentStep,
  updateCount,
}) => {
  useEffect(() => {
    updateCount(users.length);
  }, [updateCount, users]);
  return (
    <tbody>
      {users.slice(currentStep - STEP, currentStep).map((user, i) => (
        <MainTableRow
          rowID={`row${i}`}
          key={`row-${i}`}
          user={user}
          activeRow={activeRow}
          setActiveRow={setActiveRow}
        />
      ))}
    </tbody>
  );
};

MainTableBody.propTypes = {
  users: PropTypes.arrayOf(PropTypes.object).isRequired,
  activeRow: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
  setActiveRow: PropTypes.func.isRequired,
  currentStep: PropTypes.number.isRequired,
  updateCount: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  users: getFilteredUsers(state),
  activeRow: state[NameSpace.INTERFACE_DATA].activeRow,
  currentStep: state[NameSpace.USERS_DATA].currentStep,
});

const mapDispatchToProps = (dispatch) => ({
  setActiveRow(activeRow) {
    dispatch(ActionCreator.updateActiveRow(activeRow));
  },
  updateCount(count) {
    dispatch(ActionCreator.updateUserCount(count));
  },
});

export { MainTableBody };
export default connect(mapStateToProps, mapDispatchToProps)(MainTableBody);
