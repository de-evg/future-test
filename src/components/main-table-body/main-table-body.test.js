import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { MainTableBody } from "./main-table-body";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

const user = {
  id: 2,
  firstName: `Petr`,
  lastName: `Petrov`,
  email: `petr@mail.ru`,
  phone: `89105507671`,
  address: {
    streetAddress: "9792 Mattis Ct",
    city: "Waukesha",
    state: "WI",
    zip: "22178",
  },
  description: "et lacus magna dolor...",
};

it(`Render MainTableBody correctly`, () => {
  const wrapper = shallow(
    <MainTableBody
      users={[user, user, user]}
      activeRow={"row1"}
      setActiveRow={() => {}}
      currentStep={50}
      updateCount={() => {}}
    />
  );
  const tree = shallowToJson(wrapper);
  expect(tree).toMatchSnapshot();
});
