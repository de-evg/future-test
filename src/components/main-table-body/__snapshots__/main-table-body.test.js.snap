// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MainTableBody correctly 1`] = `
<tbody>
  <MainTableRow
    activeRow="row1"
    key="row-0"
    rowID="row0"
    setActiveRow={[Function]}
    user={
      Object {
        "address": Object {
          "city": "Waukesha",
          "state": "WI",
          "streetAddress": "9792 Mattis Ct",
          "zip": "22178",
        },
        "description": "et lacus magna dolor...",
        "email": "petr@mail.ru",
        "firstName": "Petr",
        "id": 2,
        "lastName": "Petrov",
        "phone": "89105507671",
      }
    }
  />
  <MainTableRow
    activeRow="row1"
    key="row-1"
    rowID="row1"
    setActiveRow={[Function]}
    user={
      Object {
        "address": Object {
          "city": "Waukesha",
          "state": "WI",
          "streetAddress": "9792 Mattis Ct",
          "zip": "22178",
        },
        "description": "et lacus magna dolor...",
        "email": "petr@mail.ru",
        "firstName": "Petr",
        "id": 2,
        "lastName": "Petrov",
        "phone": "89105507671",
      }
    }
  />
  <MainTableRow
    activeRow="row1"
    key="row-2"
    rowID="row2"
    setActiveRow={[Function]}
    user={
      Object {
        "address": Object {
          "city": "Waukesha",
          "state": "WI",
          "streetAddress": "9792 Mattis Ct",
          "zip": "22178",
        },
        "description": "et lacus magna dolor...",
        "email": "petr@mail.ru",
        "firstName": "Petr",
        "id": 2,
        "lastName": "Petrov",
        "phone": "89105507671",
      }
    }
  />
</tbody>
`;
